pilot/cmd/pilot-agent/main.go

```diff
@@ -226,6 +228,29 @@ var (
 			if err != nil {
 				return err
 			}
+
+			var tokenManager stsservice.TokenManager
+			// If security token service (STS) port is not zero, start STS server and
+			// listen on STS port for STS requests. For STS, see
+			// https://tools.ietf.org/html/draft-ietf-oauth-token-exchange-16.
+			if stsPort > 0 {
```

Comment:
Why do I need stsPort for this to work? I shouldn't need to start a local http server just to send an http request to the tokenexchanger

Reply:
The stsPort is in the existing implementation (not part of this PR) and is used by stsServer. stsServer is used in the token exchange flow for telemetry (Envoy -> stsServer), which goes from Envoy filter instead of the XDS proxy.

Reply:
I understand the existing flow is envoy --> agent -> STS, which is communicating accross processes. But for this case, we are just going to agent -> STS, so I don't think it makes sense to require spinning up a full HTTP server just for this.

Also, it appears if STSPORT=0 and XDS_ENABLE_TOKEN_FETCH_EXCHANGE=true things may be broken entirely?

See also https://github.com/istio/istio/issues/29753

Reply:
The HTTP server is not for the XDS token exchange flow; the HTTP server is for the telemetry token exchange flow. 
For telemetry token exchange flow, the token exchange goes from Envoy to STS server without going through XDS proxy. https://github.com/istio/proxy/blob/7bb04a3839ec9ae5ec77ffbb198a1177bf16b84a/extensions/stackdriver/common/utils.cc#L41 is the place where google grpc service is configured for telemetry sts token exchange.

Reply:
> The HTTP server is not for the XDS token exchange flow;

I agree, but in this PR, as far as I can tell, it is now *required* for XDS token exchange, even though its not needed

Reply:
Thanks! I have revised the PR to make the creation of token manager independent of sts port. Please take a look.